Detectors/Base/include/DetectorsBase/Detector.h

```diff
@@ -142,6 +145,26 @@ class Detector : public FairDetector
 
     ClassDefOverride(Detector, 1) // Base class for ALICE Modules
 };
+
+/// utility function to demangle cxx type names
+inline std::string demangle(const char* name)
+{
+  int status = -4; // some arbitrary value to eliminate compiler warnings
+  std::unique_ptr<char, void (*)(void*)> res{ abi::__cxa_demangle(name, nullptr, nullptr, &status), std::free };
```

Comment:
Does this work with compilers other than gcc, e.g. ICC?

Reply:
Don't know. We don't have icc. It works for our supported setup (see PR checks).

Reply:
Thinking again about this, I think it would be way better to do the proper fix and move HitType inside the detector class.

Reply:
This can be done but is orthogonal to the dictionary problem. I will track it in a separate issue.

Reply:
Ok, thanks.